src/main/webapp/app/components/timeSeries/TimeSeries.tsx

```diff
@@ -0,0 +1,186 @@
+import { APP_HISTORY_FORMAT, APP_TIME_FORMAT } from 'app/config/constants/constants';
+import React, { useEffect, useRef, useState } from 'react';
+import { Timeline, TimelineEvent } from 'react-event-timeline';
+import { TextFormat } from 'react-jhipster';
+import { Col, Row } from 'reactstrap';
+import './time-series.scss';
+import { groupTimeSeriesDataByDay } from './time-series-utils';
+
+export type RequiredTimeSeriesEventData = {
+  createdAt: Date;
+  editBy: string;
+  operation: string;
+  bubbleColor: string;
+  content: React.ReactNode;
+};
+
+export type ExtraTimeSeriesEventData = RequiredTimeSeriesEventData & {
```

Comment:
Why not merge these two into `RequiredTimeSeriesEventData`?

Reply:
I need the two to be separate so I can say if `groupByDay` is `true` or `undefined` (since it defaults to true), expect `ExtraTimeSeriesEventData` which includes `title` and subtitle`. Otherwise, just using `RequiredTimeSeriesData` which does not include those fields (since the non-grouped view does not use the title and subtitle)